File: src/test/java/org/geepawhill/contentment/test/PointPairAssert.java
package org.geepawhill.contentment.test;

import static org.geepawhill.contentment.test.ContentmentAssertions.assertThat;

import org.assertj.core.api.AbstractAssert;
import org.geepawhill.contentment.geometry.PointPair;

public class PointPairAssert extends AbstractAssert<PointPairAssert, PointPair>
{
	public PointPairAssert(PointPair actual)
	{
		super(actual, PointPairAssert.class);
	}

	public PointPairAssert isEqualTo(Object expected)
	{
		if(!(expected instanceof PointPair)) failWithMessage("Not a PointPair");
		PointPair expectedPair = (PointPair) expected;
		assertThat(expectedPair.getFrom()).isEqualTo(actual.getFrom());
		assertThat(expectedPair.getTo()).isEqualTo(actual.getTo());
		return this;
	}
}

File: src/test/kotlin/org/geepawhill/contentment/test/PointPairAssert.kt
package org.geepawhill.contentment.test

import org.assertj.core.api.AbstractAssert
import org.geepawhill.contentment.geometry.PointPair
import org.geepawhill.contentment.test.ContentmentAssertions.Companion.assertThat

class PointPairAssert(actual: PointPair) : AbstractAssert<PointPairAssert, PointPair>(actual, PointPairAssert::class.java) {

    override fun isEqualTo(expected: Any): PointPairAssert {
        if (expected !is PointPair) failWithMessage("Not a PointPair")
        val expectedPair = expected as PointPair
        assertThat(expectedPair.from).isEqualTo(actual.from)
        assertThat(expectedPair.to).isEqualTo(actual.to)
        return this
    }
}
